Compilation fails on Haiku

Gerbera 1.8.2 fails here:

```
/Ports/haikuports> uname -a
Haiku Flatty 1 hrev55120 May 31 2021 06:02:18 x86_64 x86_64 Haiku
```

```
-- The CXX compiler identification is GNU 11.1.0
-- The C compiler identification is GNU 11.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Gerbera Version: 1.8.2
-- Looking for C++ include optional
-- Looking for C++ include optional - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Looking for __atomic_fetch_add_4 in atomic
-- Looking for __atomic_fetch_add_4 in atomic - found
-- Looking for C++ include filesystem
-- Looking for C++ include filesystem - found
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED - Success
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Failed
-- Found Iconv: /boot/system/develop/lib/libiconv.so  
-- Looking for BSD native UUID
-- Looking for libuuid
-- Looking for libuuid - found
-- Found libuuid: TRUE  
-- Looking for native LFS support
-- Looking for native LFS support - found
-- Found LFS: TRUE  
-- Found spdlog: /boot/system/lib/cmake/spdlog/spdlogConfig.cmake (found version "1.8.0") 
-- Found pugixml: /boot/system/develop/lib/libpugixml.so  
-- Found SQLite3: /boot/system/develop/headers (found suitable version "3.34.1", minimum required is "3.7.0") 
-- Found duktape: /boot/system/develop/lib/libduktape.so  
-- Found TagLib: -L/packages/taglib-1.12-1/.self/develop/lib -ltag -lz
-- Found LibMagic: /packages/file-5.40-1/.self/develop/lib/libmagic.so  
-- Checking for modules 'libavformat;libavutil'
--   Found libavformat, version 58.29.100
--   Found libavutil, version 56.31.100
-- Found EXIF: /boot/system/develop/lib/libexif.so (found version "0.6.21") 
-- Found EBML: /packages/libebml-1.4.1-1/.self/develop/lib/libebml.so  
-- Found Matroska: /packages/libmatroska-1.6.2-1/.self/develop/lib/libmatroska.so  
-- Found ZLIB: /boot/system/develop/lib/libz.so (found version "1.2.11") 
-- Configuring done
-- Generating done
-- Build files have been written to: /sources/gerbera-1.8.2/build
[  2%] Building CXX object CMakeFiles/libgerbera.dir/src/action_request.cc.o
[  4%] Building CXX object CMakeFiles/libgerbera.dir/src/cds_resource.cc.o
[  4%] Building CXX object CMakeFiles/libgerbera.dir/src/cds_objects.cc.o
[  4%] Building CXX object CMakeFiles/libgerbera.dir/src/config/client_config.cc.o
In file included from /sources/gerbera-1.8.2/src/util/timer.h:41,
                 from /sources/gerbera-1.8.2/src/content/autoscan.h:40,
                 from /sources/gerbera-1.8.2/src/content/content_manager.h:41,
                 from /sources/gerbera-1.8.2/src/config/client_config.cc:28:
/sources/gerbera-1.8.2/src/util/thread_runner.h: In member function 'void ThreadRunner<Condition, Mutex>::startThread()':
/sources/gerbera-1.8.2/src/util/thread_runner.h:187:13: error: there are no arguments to 'pthread_attr_setinheritsched' that depend on a template parameter, so a declaration of 'pthread_attr_setinheritsched' must be available [-fpermissive]
  187 |             pthread_attr_setinheritsched(attr, PTHREAD_EXPLICIT_SCHED);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sources/gerbera-1.8.2/src/util/thread_runner.h:187:13: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
[  5%] Building CXX object CMakeFiles/libgerbera.dir/src/config/config_definition.cc.o
/sources/gerbera-1.8.2/src/util/thread_runner.h: In instantiation of 'void ThreadRunner<Condition, Mutex>::startThread() [with Condition = std::condition_variable; Mutex = std::mutex]':
/sources/gerbera-1.8.2/src/util/thread_runner.h:178:10:   required from here
/sources/gerbera-1.8.2/src/util/thread_runner.h:187:41: error: 'pthread_attr_setinheritsched' was not declared in this scope; did you mean 'pthread_attr_setguardsize'?
  187 |             pthread_attr_setinheritsched(attr, PTHREAD_EXPLICIT_SCHED;
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |             pthread_attr_setguardsize
CMakeFiles/libgerbera.dir/build.make:117: recipe for target 'CMakeFiles/libgerbera.dir/src/config/client_config.cc.o' failed
make[2]: *** [CMakeFiles/libgerbera.dir/src/config/client_config.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /sources/gerbera-1.8.2/src/util/timer.h:41,
                 from /sources/gerbera-1.8.2/src/content/autoscan.h:40,
                 from /sources/gerbera-1.8.2/src/config/config_setup.h:38,
                 from /sources/gerbera-1.8.2/src/config/config_definition.cc:35:
/sources/gerbera-1.8.2/src/util/thread_runner.h: In member function 'void ThreadRunner<Condition, Mutex>::startThread()':
/sources/gerbera-1.8.2/src/util/thread_runner.h:187:13: error: there are no arguments to 'pthread_attr_setinheritsched' that depend on a template parameter, so a declaration of 'pthread_attr_setinheritsched' must be available [-fpermissive]
  187 |             pthread_attr_setinheritsched(attr, PTHREAD_EXPLICIT_SCHED);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sources/gerbera-1.8.2/src/util/thread_runner.h:187:13: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/sources/gerbera-1.8.2/src/util/thread_runner.h: In instantiation of 'void ThreadRunner<Condition, Mutex>::startThread() [with Condition = std::condition_variable; Mutex = std::mutex]':
/sources/gerbera-1.8.2/src/util/thread_runner.h:178:10:   required from here
/sources/gerbera-1.8.2/src/util/thread_runner.h:187:41: error: 'pthread_attr_setinheritsched' was not declared in this scope; did you mean 'pthread_attr_setguardsize'?
  187 |             pthread_attr_setinheritsched(attr, PTHREAD_EXPLICIT_SCHED;
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |             pthread_attr_setguardsize
CMakeFiles/libgerbera.dir/build.make:131: recipe for target 'CMakeFiles/libgerbera.dir/src/config/config_definition.cc.o' failed
make[2]: *** [CMakeFiles/libgerbera.dir/src/config/config_definition.cc.o] Error 1
CMakeFiles/Makefile2:110: recipe for target 'CMakeFiles/libgerbera.dir/all' failed
make[1]: *** [CMakeFiles/libgerbera.dir/all] Error 2
Makefile:155: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/src/main.cc
+++ b/src/main.cc
@@ -261,7 +261,7 @@ int main(int argc, char** argv, char** envp)
 // mac os x does this differently, setgid and setuid are basically doing the same
 // as setresuid and setresgid on linux: setting all of real{u,g}id, effective{u,g}id and saved-set{u,g}id
 // Solaroid systems are likewise missing setresgid and setresuid
-#if defined(__APPLE__) || defined(SOLARIS) || defined(__CYGWIN__)
+#if defined(__APPLE__) || defined(SOLARIS) || defined(__CYGWIN__) || defined(__HAIKU__)
             // set group-ids, then add. groups, last user-ids, all need to succeed
             if (0 != setgid(user_id->pw_gid) || 0 != initgroups(user_id->pw_name, user_id->pw_gid) || 0 != setuid(user_id->pw_uid)) {
 #else

--- a/src/util/thread_runner.h
+++ b/src/util/thread_runner.h
@@ -177,9 +177,10 @@ protected:
     /// \brief start the thread
     void startThread() override
     {
-#ifndef SOLARIS
+#if !defined(SOLARIS) && !defined(__HAIKU__)
         // default scoping on solaroid systems is in fact PTHREAD_SCOPE_SYSTEM
         // plus, setting PTHREAD_EXPLICIT_SCHED requires elevated priveleges
+		// while Haiku doesn't implements pthread_attr_setinheritsched yet
         if (config->getBoolOption(CFG_THREAD_SCOPE_SYSTEM)) {
             attr = new pthread_attr_t;
             pthread_attr_init(attr);

--- a/src/util/thread_runner.h
+++ b/src/util/thread_runner.h
@@ -180,7 +180,7 @@ protected:
 #if !defined(SOLARIS) && !defined(__HAIKU__)
         // default scoping on solaroid systems is in fact PTHREAD_SCOPE_SYSTEM
         // plus, setting PTHREAD_EXPLICIT_SCHED requires elevated priveleges
-		// while Haiku doesn't implements pthread_attr_setinheritsched yet
+        // while Haiku doesn't implements pthread_attr_setinheritsched yet
         if (config->getBoolOption(CFG_THREAD_SCOPE_SYSTEM)) {
             attr = new pthread_attr_t;
             pthread_attr_init(attr);

